client/src/components/CommentsContainer/index.tsx

```diff
@@ -1,27 +1,159 @@
+/* eslint-disable no-useless-escape */
 import './style.css'
 
-import { FC } from 'react'
+import React, { FC, useState } from 'react'
 import {
-  Button,
+  Button, Alert, AlertTitle, CircularProgress,
 } from '@mui/material'
+// import AddOutlinedIcon from '@mui/icons-material/AddOutlined'
 
-import AddOutlinedIcon from '@mui/icons-material/AddOutlined'
+import { useParams } from 'react-router-dom'
+import InfiniteScroll from 'react-infinite-scroll-component'
 import Comment from '../Comment'
+import ApiService from '../../services/ApiService'
+import {
+  IComments,
+  IOneComment,
+} from '../../interfaces'
+import AddCommentModal from '../AddCommentsModal'
+// eslint-disable-next-line no-undef
+const CommentsContainer:FC = (): JSX.Element => {
+  const initCommentsValue: IComments = {
+    message: '',
+    data: [
+      {
+        id: 2,
+        content: '',
+        image: '',
+        createdAt: '',
+        updatedAt: '',
+        UserId: 1,
+        EventId: 1,
+        User: {
+          id: 1,
+          username: '',
+          profileImg: '',
+        },
+      },
+    ],
+  }
+
+  const initOneCommentValue: IOneComment = {
+    message: '',
+    data:
+      {
+        id: 0,
+        content: '',
+        image: '',
+        createdAt: '',
+        updatedAt: '',
+        UserId: 1,
+        EventId: 1,
+        User: {
+          id: 1,
+          username: '',
+          profileImg: '',
+        },
+      },
+
+  }
+
+  const [comments, setComments] = useState<IComments>(initCommentsValue)
+  const [newComment, setNewComments] = useState<IOneComment>(initOneCommentValue)
+  const [showMore, setShowMore] = useState<number>(1)
+  const [allComments, setAllComments] = useState<IComments[]>([])
+  const [hasMore, setHasMore] = useState<boolean>(true)
+  const [open, setOpen] = useState<boolean>(false)
+  const [error, setError] = useState<boolean | string>(false)
+  const [loader, setLoader] = useState<boolean>(true)
+
+  const handleOpen = ():void => setOpen(true)
+  const handleClose = ():void => setOpen(false)
+  const handleShowMore = ():void => setShowMore(showMore + 1)
+
+  const idParams = useParams().id
+
+  React.useEffect(() => {
+    (async (): Promise<void> => {
+      try {
+        const result = await ApiService.get(`/events/${idParams}/comments?offset=${showMore}`)
+        setComments(result.data)
+        setAllComments([...allComments, ...result.data.data])
+        if (!result.data.data.length) setHasMore(false)
+        setLoader(false)
+      } catch (err) {
+        setError('We\'re having some errors in getting the information. We\'re working on it.')
```

Comment:
the message here should be a global one
i noticed you hard write it in many places